com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/ManifestBuilder.java

```diff
@@ -284,15 +287,12 @@ public static PublicKey loadPublicKey(String filepath, SignAlgorithm algorithm)
     private String privateKeyFilepath = null;
     private String publicKeyFilepath = null;
     private String projectIdentifier = null;
-    private ResourceNode root = null;
+    private ResourceGraph resourceGraph = null;
     private boolean outputManifestHash = false;
     private byte[] manifestDataHash = null;
     private byte[] archiveIdentifier = new byte[ArchiveBuilder.MD5_HASH_DIGEST_BYTE_LENGTH];
-    private Set<String> excludedResources = new HashSet<>();
-    private HashMap<String, ResourceNode> pathToNode = new HashMap<>();
-    private HashMap<String, List<String>> pathToDependants = new HashMap<>();
+    private HashMap<String, HashSet<String>> pathToDependants = new HashMap<>();
```

Comment:
Using of a HashSet instead of a List helps to make sure that all the resource entries are unique